### PR TITLE
fix: Prevent exiting the page if there are unsaved changes to the default props

### LINKED ISSUE #2495: Prevent exiting the page if there are unsaved changes to the default props
If the Remotion Studio Props Editor has unsaved changes, and you are about to reload, then an alert should be triggered using `window.onbeforeunload` telling the user that there are unsaved changes.

To determine if there are unsaved changes, check only if the root object is changed.

Once the changes are saved, the handler should be removed.

**Special case:** 

There are 5 cases where Remotion Studio will reload itself (look for `window.location.reload()`):

- New input props or env variables (packages/cli/src/event-source.ts) ❌
- Fatal React error (packages/cli/src/preview-server/error-overlay/entry-basic.ts) ✅
- Hook order changed (packages/cli/src/preview-server/error-overlay/react-overlay/listen-to-runtime-errors.ts) ❌
- Failed fast refresh (packages/cli/src/preview-server/hot-middleware/process-update.ts) ❌

✅ The error is unrecoverable and we anyway lost our changes, so we can refresh
❌ The error is not recoverable, and we should not refresh, instead show a notification using `notificationCenter.current?.addNotification`

💎 This issue has a bounty on it!

Read our contributing guidelines: 
- **How to contribute**: https://www.remotion.dev/docs/contributing/
- **Implementing a new feature**: https://www.remotion.dev/docs/contributing/feature
- **Implementing a new option**: https://www.remotion.dev/docs/contributing/option
- **Write documentation**: https://www.remotion.dev/docs/contributing/docs
- **Bounty information**: https://www.remotion.dev/docs/contributing/bounty

/bounty 100

### PR DESCRIPTION
**Pull request by rishi-raj-jain**
fix #2495
/claim #2495

### CODE REVIEW

**Comment by rishi-raj-jain** [comment]
@JonnyBurger 

Can you give the message and duration you expect to set on this?

I added the event listener in the render modal, let me know if you think I shall shift it to somewhere else.

File: packages/cli/src/event-source.ts [hunk 1]
```diff
@@ -32,7 +35,16 @@ export const openEventSource = () => {
 			newEvent.type === 'new-input-props' ||
 			newEvent.type === 'new-env-variables'
 		) {
-			window.location.reload();
+			if (window.unsavedProps) {
+				notificationCenter.current?.addNotification({
+					id: 'random',
+					content: 'Do not save hey',
```
Comment (JonnyBurger) [inline packages/cli/src/event-source.ts]: - The content could be "There are unsaved changes to the props. Do you want to leave?"
- The notification center will do nothing here, you can only prevent a page from closing if you use `window.oneforeunload`

Example:
```
window.addEventListener('beforeunload', (event) => {
  event.returnValue = 'Are you sure you want to leave?';
});
```

File: packages/cli/src/editor/helpers/document-title.ts [hunk 2]
```diff
@@ -6,12 +6,22 @@ let unsavedProps = false;
 let tabInactive = false;
 let renderJobs: RenderJob[] = [];
 
+declare global {
```
Comment (JonnyBurger) [inline packages/cli/src/editor/helpers/document-title.ts]: All global variables are defined in core/src/index.ts.
Can you move it there as well and update the variable name with a `remotion_` prefix?

**Review by JonnyBurger — CHANGES_REQUESTED** [review]

**Comment by JonnyBurger** [comment]
Thanks!

**Review by JonnyBurger — APPROVED** [review]

**Pull request merged**